#ifndef PARADIALOG_H
#define PARADIALOG_H
#include "global_defines.h"
#include "settingfile.h"
#include "qmessagebox.h"
#include "qfiledialog.h"
#include <QDialog>
#include <QMessageBox>

const double Factor = 150;                    //光速/2
const int SIZE_OF_FILE_HEADER = 6;			  //单个单文件头的大小,单位：字节
//const int DATA_MEMORY = 64*1000*1000;		  //采集卡内存空间（点）
//const int UPLOAD_SPEED = 20;				  //上传速度MB/s

namespace Ui {
class paraDialog;
}

class paraDialog : public QDialog
{
    Q_OBJECT

public:
    explicit paraDialog(QWidget *parent = 0);
    ~paraDialog();

    void init_setting(const ACQSETTING &setting, bool sop);
    void initial_para();
    ACQSETTING get_settings(void);

public slots:
     void on_checkBox_autoCreate_DateDir_clicked();

private slots:
    //激光参数
    void set_laserEP();
    void set_laserRPF();
    void set_laserPulseWidth();
    void set_laserWaveLength();
    void set_AOM_Freq();
    //扫描参数
    void set_elevationAngle();
    void set_start_azAngle();
    void set_step_azAngle();
    void set_circleNum();
    void set_angleNum();
    void set_anglekey();
    void set_circlekey();
    void set_detectMode();
//    void set_SP_Interval();
    void set_motorSP();
    void set_time_direct_interval();
    void set_time_circle_interval();
    //采样设置
    void set_sampleFreq();
    void set_plsAccNum();
    void set_velocity_band();
    void set_Trigger_Level();
    void set_nRangeBin();
    void set_nPointsPerBin();
    void set_nDir_VectorCal();

    //文件存储
    void on_pushButton_pathModify_clicked();
    void set_nMaxDir_infile();

    void on_pushButton_save_clicked();

    void on_pushButton_cancel_clicked();

    void on_pushButton_load_clicked();

    void on_pushButton_reset_clicked();

    void on_pushButton_sure_clicked();

    void on_comboBox_DetetectMode_currentIndexChanged(int index);

//    void show_detect_mode();                    //参考信息――探测方式
    void show_deteRange();                       //探测距离
    void show_RangeReso();                       //距离分辨率



    void on_comboBox_lasermode_currentIndexChanged(int index);

private:
    Ui::paraDialog *ui;

    ACQSETTING psetting;
    ACQSETTING defaulsetting;
    settingfile dlg_setfile;
    QString profile_path;                      //配置文件路径

//    double triggerlevel;                      //触发电压
    double deteRange;
    double filesize;
    double velodatafilesize;		           //单方向上的数据量
    double specdatafilesize;
    double RangeReso;                          //距离分辨率
//    int sampleNum;			               //单个脉冲的采样点数
    bool nocollecting;		                   //是否正在采集数据
//    int pagePerPls;			                   //单个脉冲的page数
    QString lenStr;

    void filesize_over();				         //判断单文件量超过采集卡内存
    void show_DatafilePath(QString str);
    quint64 getDiskFreeSpace(QString driver);	 //获取路径下的硬盘剩余空间大小

    void update_show();
//    void show_detect_mode();                     //参考信息――探测方式
//    void show_deteRange();                       //探测距离
//    void show_RangeReso();                       //距离分辨率
    void show_filesize();

};

#endif // PARADIALOG_H
